bot/exts/holidays/easter/egghead_quiz.py

```diff
@@ -97,7 +121,11 @@ async def eggquiz(self, ctx: commands.Context) -> None:
     @staticmethod
     async def already_reacted(message: discord.Message, user: Union[discord.Member, discord.User]) -> bool:
         """Returns whether a given user has reacted more than once to a given message."""
-        users = [u.id for reaction in [await r.users().flatten() for r in message.reactions] for u in reaction]
+        users = []
+        for reaction in message.reactions:
+            async for usr in reaction.users():
+                users.append(usr.id)
+
         return users.count(user.id) > 1  # Old reaction plus new reaction
```

Comment:
Time complexity in this part seems like it can be better, currently iterates for 2N times for N total users reacted. For 2 reactions, each with 500 users, currently, we iterate a constant of 2000 times (for loop that builds list + list.count) on each new reaction.

Suggest this as an alternative:
In the same case of 1000 users, this would iterate a best case of 1 time and a worst case of 500 times.
```py
@staticmethod
async def already_reacted(new_reaction: discord.Reaction, user: Union[discord.Member, discord.User]) -> bool:
    """Returns whether a given user has reacted more than once to a given message."""
    message = new_reaction.message
    for reaction in message.reactions:
        if reaction.emoji == new_reaction.emoji:
            continue  #  Skip new reaction, duplicate impossible
        async for usr in reaction.users():
            # Duplicate if user id matches anywhere else
            if usr.id == user.id:
                return True
    return False
```

Reply:
Done, thanks :+1: